src/ar.erl

```diff
@@ -236,13 +236,22 @@ start(
 		disable = Disable,
 		content_policy_files = ContentPolicyFiles,
 		transaction_blacklist_files = TransactionBlacklistFiles,
-		gateway = GatewayOpts,
-		custom_domains = GatewayCustomDomains,
+		gateway_domain = GatewayDomain,
+		gateway_custom_domains = GatewayCustomDomains,
 		requests_per_minute_limit = RequestsPerMinuteLimit
 	}) ->
 	%% Start the logging system.
 	error_logger:logfile({open, Filename = generate_logfile_name()}),
 	error_logger:tty(false),
+	%% Verify port collisions when gateway enabled
+	GatewayOpts = gateway_opts(GatewayDomain, GatewayCustomDomains),
```

Comment:
Looks like `gateway_opts` are not used anywhere else, could we perhaps just check if `GatewayDomain` is set here?

Reply:
I'm not sure I get your feedback, the `gateway_opts` helper function is just used once here to avoid inlining more login into the already huge start function, although I could inline it if you insist, but the `GatewayOpts` variable is also used later below

Reply:
May be it's cleaner if we keep it like this, let's keep it.